internal/controller/operatorconfigmap_controller.go

```diff
@@ -890,3 +784,79 @@ func (c *OperatorConfigMapReconciler) deleteDelegatedCSI() error {
 	}
 	return nil
 }
+
+func (c *OperatorConfigMapReconciler) reconcileSubscriptionLabelForWebhook() error {
+	sub, err := c.getSubscriptionByPackageName("ocs-client-operator")
+	if err != nil {
+		return err
+	}
+	if utils.AddLabel(sub, subscriptionLabelKey, subscriptionLabelValue) {
+		if err := c.update(sub); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *OperatorConfigMapReconciler) reconcileSubscriptions() error {
+	pkgNames := [][2]string{
+		// d/s and u/s names of packages for fallback
+		{"ocs-client-operator", ""},
+		{"odf-csi-addons-operator", "csi-addons"},
+		{"cephcsi-operator", "ceph-csi-operator"},
+		{"recipe", ""},
+	}
+	if c.getNoobaaSubManagementConfig() {
+		pkgNames = append(pkgNames, [2]string{"mcg-operator", "noobaa-operator"})
+	}
+	installPlansForApproval := map[string]struct{}{}
+	subsMissingInstallPlans := false
+	for _, pkgName := range pkgNames {
+		sub, err := c.getSubscriptionByPackageName(pkgName[0])
+		if pkgName[1] != "" && kerrors.IsNotFound(err) {
+			sub, err = c.getSubscriptionByPackageName(pkgName[1])
+		}
+		if err != nil {
+			return err
+		}
+		if c.subscriptionChannel != "" && c.subscriptionChannel != sub.Spec.Channel {
+			sub.Spec.Channel = c.subscriptionChannel
+			if err := c.update(sub); err != nil {
+				return fmt.Errorf("failed to update subscription channel of %s to %v: %v", sub.Name, c.subscriptionChannel, err)
+			}
+		}
+		if sub.Spec.InstallPlanApproval == opv1a1.ApprovalManual && sub.Status.CurrentCSV != sub.Status.InstalledCSV {
+			ipRef := sub.Status.InstallPlanRef
+			if ipRef != nil {
+				installPlansForApproval[ipRef.Name] = struct{}{}
+			} else {
+				subsMissingInstallPlans = true
+				c.log.Info("installplan is not found in subscription %s status", sub.Name)
+			}
+		}
+	}
+
+	approvePatch := client.RawPatch(types.MergePatchType, []byte(`{"spec":{"approved":true}}`))
+	for ipName := range installPlansForApproval {
+		installPlan := &opv1a1.InstallPlan{}
+		installPlan.Name = ipName
+		installPlan.Namespace = c.OperatorNamespace
+		if err := c.get(installPlan); client.IgnoreNotFound(err) != nil {
+			return err
+		} else if installPlan.UID == "" {
+			// if installplan is not found but mentioned in subscription then it's deleted and
+			// we can't do anything in that scenario
+			c.log.Info("Warning! installplan %s not found, probably deleted", installPlan.Name)
+			continue
+		}
+
+		if err := c.Patch(c.ctx, installPlan, approvePatch); err != nil {
+			return err
+		}
+	}
+	if subsMissingInstallPlans {
+		// requeue for OLM to update installplan in the subscriptions
+		return fmt.Errorf("one or more subscriptions are missing installplan")
+	}
```

Comment:
I think this is wrong. We do not control the existence of install plan. If OLM decided that an install plan is not needed or a user manually deleted it  I don't think we should error the entire reconcile